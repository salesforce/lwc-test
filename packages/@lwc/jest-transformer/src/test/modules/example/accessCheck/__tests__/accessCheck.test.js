	/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */
import { createElement } from 'lwc';
import AccessCheck from 'example/accessCheck';

jest.mock(
    '@salesforce/accessCheck/mocked',
    () => {
        return { default: 'value set in test' };
    },
    { virtual: true }
);

afterEach(() => {
    while (document.body.firstChild) {
        document.body.removeChild(document.body.firstChild);
    }
});

describe('example-access-check', () => {
    it('default snapshot', () => {
        const element = createElement('example-access-check', { is: AccessCheck });
        document.body.appendChild(element);
        expect(element).toMatchSnapshot();
    });

    it('returns default permission value as import path', () => {
        const element = createElement('example-access-check', { is: AccessCheck });
        document.body.appendChild(element);
        const accessCheckValue = document.body
            .querySelector('example-access-check')
            .shadowRoot.querySelector('.unmockedPermission').textContent;
        expect(accessCheckValue).toBe('unmocked');
    });

    it('returns value from mock defined in test file', () => {
        const element = createElement('example-access-check', { is: AccessCheck });
        document.body.appendChild(element);
        const accessCheckValue = document.body
            .querySelector('example-access-check')
            .shadowRoot.querySelector('.mockedPermission').textContent;
        expect(accessCheckValue).toBe('value set in test');
    });
});