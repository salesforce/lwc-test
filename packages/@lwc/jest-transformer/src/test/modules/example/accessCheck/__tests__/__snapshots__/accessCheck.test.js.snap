// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example-access-check default snapshot 1`] = `
<example-access-check>
  #shadow-root(open)
    <p
      class="mockedPermission"
    >
      value set in test
    </p>
    <p
      class="unmockedPermission"
    >
      unmocked
    </p>
</example-access-check>
`;